datalad_neuroimaging/bidsapp.py

```diff
@@ -0,0 +1,98 @@
+import logging
+from os.path import join as opj
+from datalad.interface.base import build_doc, Interface
+from datalad.support.param import Parameter
+from datalad.distribution.dataset import datasetmethod, EnsureDataset, \
+    require_dataset, resolve_path, Dataset
+from datalad.interface.utils import eval_results
+from datalad.support.constraints import EnsureStr
+from datalad.support.constraints import EnsureNone
+from datalad.support.exceptions import InsufficientArgumentsError
+from datalad.coreapi import run
+from datalad.interface.results import get_status_dict
+
+
+lgr = logging.getLogger("datalad.cbbsimaging.bidsapp")
+
+
+@build_doc
+class BidsApp(Interface):
+    """Run a BIDSApp on the dataset
+    """
+
+    _params_ = dict(
+        dataset=Parameter(
+            args=("-d", "--dataset"),
+            doc="""dataset to run the BIDSApp on""",
+            constraints=EnsureDataset() | EnsureNone()),
+        cmd=Parameter(
+            args=("cmd",),
+            doc="""Preconfigured BIDSApp to run. The name corresponds to
+            datalad.neuroimaging.bidsapp.NAME configuration.""",
+            metavar="CMD",
+            constraints=EnsureStr() | EnsureNone())
+        )
+
+    @staticmethod
+    @datasetmethod(name='run_bidsapp')
+    @eval_results
+    def __call__(cmd, dataset=None):
+
+        dataset = require_dataset(dataset, check_installed=True,
+                                  purpose="run BIDS App")
+
+        container_name = \
+            dataset.config.get("datalad.neuroimaging.bidsapp.%s.container-name"
+                               % cmd)
+        if not container_name:
+            raise ValueError("Missing configuration for "
+                             "datalad.neuroimaging.bidsapp.%s.container-name"
+                             % cmd)
+
+        container_ds = Dataset(opj(dataset.path, ".git",
+                                   "environment", container_name))
```

Comment:
I think it is better to put it under .git/datalad/environment (or some other name, but inside a datalad namespace), otherwise we are in potential conflict with other git-based tools.

Reply:
Right.